### PR TITLE
Update tty tools

### PR DESCRIPTION
**Pull request by heiner**

### CODE REVIEW

File: nle/scripts/read_tty.py [hunk 1]
```diff
@@ -49,42 +73,129 @@ def getfile(filename):
...
-                    datetime.datetime.fromtimestamp(timestamp),
-                    color(channel, 2),
-                    color("{", 11),
-                    data,
-                    color("}", 11),
+                char, *_ = struct.unpack("<B", data)
+                data = chr(char).encode("ascii", "backslashreplace")
+                arrow = "->"
+
+            data = str(data)[2:-1]  # Strip b' and '
```
Comment (condnsdmatters) [inline nle/scripts/read_tty.py:172]: probably better to use a more 'proper' way of doing this that doesnt depend on how Python decides to represent a converted bytes to string?

**Review by condnsdmatters — APPROVED** [review]

**Comment by condnsdmatters** [comment]
Im just testing it out and will add some screen shots to the PR

File: nle/scripts/read_tty.py (hunk 1 above)
Comment (heiner) [reply nle/scripts/read_tty.py:172]: I agree! But what is the proper way of doing exactly that? It's not `data.encode("ascii", "backslashreplace")` ...

Comment (heiner) [reply nle/scripts/read_tty.py:172]: StackOverflow seems to suggest doing what we are doing here ... https://stackoverflow.com/questions/54889234/how-to-decode-a-string-representation-of-a-bytes-object

**Comment by condnsdmatters** [comment]
Screenshot 2021-11-02 at 15 09 14

File: setup.py [hunk 2]
```diff
@@ -92,6 +92,8 @@ def build_extension(self, ext):
         "nle-play = nle.scripts.play:main",
         "nle-ttyrec = nle.scripts.ttyrec:main",
         "nle-ttyplay = nle.scripts.ttyplay:main",
+        "nle-ttyplay2 = nle.scripts.ttyplay2:main",
```
Comment (condnsdmatters) [inline setup.py:95]: the ttyplay2 file doesnt have a main function? or something?

I'm getti
Screenshot 2021-11-02 at 15 10 31
ng

**Comment by condnsdmatters** [comment]
Screenshot 2021-11-02 at 15 12 30

File: setup.py (hunk 2 above)
Comment (heiner) [reply setup.py:95]: Good catch; fixed.

File: nle/scripts/read_tty.py (hunk 1 above)
Comment (heiner) [reply nle/scripts/read_tty.py:172]: Looks like this is the way to do this ... https://stackoverflow.com/questions/16748083/suppress-print-without-b-prefix-for-bytes-in-python-3 and https://github.com/python/cpython/blob/e1b900247227dad49d8231f1d028872412230ab4/Objects/bytesobject.c#L1345 ...

**Pull request merged**